pkg/model/combine.go

```diff
@@ -18,12 +20,29 @@ type objectCombiner struct{}
 
 var ObjectCombiner = objectCombiner{}
 
+var _ common.ObjectCombiner = (*objectCombiner)(nil)
+
 // Combine implements tempodb/encoding/common.ObjectCombiner
-func (o objectCombiner) Combine(objA []byte, objB []byte, dataEncoding string) ([]byte, bool) {
-	combinedTrace, wasCombined, err := CombineTraceBytes(objA, objB, dataEncoding, dataEncoding)
-	if err != nil {
-		level.Error(log.Logger).Log("msg", "error combining trace protos", "err", err.Error())
+func (o objectCombiner) Combine(dataEncoding string, objs ...[]byte) ([]byte, bool) {
+	if len(objs) <= 0 {
+		return nil, false
+	}
+
+	if len(objs) == 1 {
+		return objs[0], false
 	}
+
+	combinedTrace := objs[0]
+	var wasCombined bool
+	var err error
+	for _, obj := range objs[1:] {
+		combinedTrace, wasCombined, err = CombineTraceBytes(combinedTrace, obj, dataEncoding, dataEncoding)
```

Comment:
CombineTraceBytes is just a wrapper for CombineTraceProtos. It would be more efficient to unmarshal to proto using the dataencoding here and then call CombineTraceProtos directly. Otherwise you're repeatedly marshalling/unmarshalling if len(objs) > 2

Reply:
Yep we definitely want to do this optimization, but nothing is combining more than 2 entries yet.  Places where that's needed are DedupingIterator (wal), and MultiblockIterator (compaction).   Didn't want to tackle them in this PR. Thoughts?

Reply:
That's fine. Let's add a todo to make this improvement in the future.

Reply:
Yeah there's no changes to the way objects are currently being combined, will add this todo